server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java

```diff
@@ -188,6 +202,23 @@ public Task createTask(long id, String type, String action, TaskId parentTaskId,
         }
     }
 
+    private <Params extends PersistentTaskParams> void notifyMasterOfFailedTask(PersistentTask<Params> taskInProgress, Exception e) {
+        persistentTasksService.sendCompletionRequest(taskInProgress.getId(), taskInProgress.getAllocationId(), e,
+            new ActionListener<>() {
+                @Override
+                public void onResponse(PersistentTask<?> persistentTask) {
+                    logger.trace("completion notification for task [{}] with id [{}] was successful", taskInProgress.getTaskName(),
```

Comment:
I think this trace message should include the work "failed", something like: "completion notification for failed task [{}] with id [{}] was successful"